update FAQ to explain license & more (#20)
returntocorp/semgrep-docs: docs/faq.md

# General

> #### What is the license for semgrep? What about the rules?

Reviewer: ```suggestion
#### How is Semgrep and its rules licensed?
```

Reviewer 2: ```suggestion
#### How are Semgrep and its rules licensed?
```

[Semgrep CLI](https://github.com/returntocorp/semgrep) is open-source, licensed under the [LGPL 2.1](https://tldrlegal.com/license/gnu-lesser-general-public-license-v2.1-(lgpl-2.1)) license — you can use it at work, on private and proprietary code, no problem!
    
The [Semgrep Registry](https://semgrep.dev/explore) contains rules from many different contributors, often under differing licenses. If you use Semgrep Registry through Semgrep.dev, r2c is vetting the rules to make sure that the licenses are compatible with using them at your company.

> Many of the rules that r2c writes have their source available at [returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/). These are licensed as LGPL 2.1 under the [Commons Clause](https://commonsclause.com/). This means the source for the rules is available, but they can’t be resold without r2c’s permission. Although the Commons Clause and related licenses like the BSL/SSPL are embraced by many great open-source projects ([Sentry](https://blog.sentry.io/2019/11/06/relicensing-sentry/), [Cockroach](https://www.cockroachlabs.com/blog/oss-relicensing-cockroachdb/), [Mongo](https://www.mongodb.com/licensing/server-side-public-license/faq)), it is not technically open-source under the OSI definition. Why do we have this restriction if we love open source software so much? Since r2c offers a paid, hosted version at Semgrep.dev, it’s important for us to have this restriction so other companies like major cloud providers can’t just resell the rules we’ve worked on as a competing service to Semgrep.dev.

Reviewer: ```suggestion
Many of the rules that r2c writes have their source available at [returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/). These are licensed as LGPL 2.1 under the [Commons Clause](https://commonsclause.com/). This means the source for the rules is available, but they can’t be resold without r2c’s permission. Although the Commons Clause and related licenses like the BSL/SSPL are embraced by many great open-source projects ([Sentry](https://blog.sentry.io/2019/11/06/relicensing-sentry/), [Cockroach](https://www.cockroachlabs.com/blog/oss-relicensing-cockroachdb/), [Mongo](https://www.mongodb.com/licensing/server-side-public-license/faq)), it is not technically open-source under the OSI definition. Why do we have this restriction if we love open-source software so much? Since r2c offers a paid, hosted version at Semgrep.dev, it’s important for us to have this restriction so other companies like major cloud providers can’t just resell the rules we’ve worked on as a competing service to Semgrep.dev.
```

Reviewer 2: ```suggestion
The source for many of r2c’s rules is available at [returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/). These are licensed as LGPL 2.1 under the [Commons Clause](https://commonsclause.com/). This means the source for the rules is available, but they can’t be resold without r2c’s permission. Although the Commons Clause and related licenses like the BSL/SSPL are embraced by many great open-source projects ([Sentry](https://blog.sentry.io/2019/11/06/relicensing-sentry/), [Cockroach](https://www.cockroachlabs.com/blog/oss-relicensing-cockroachdb/), [Mongo](https://www.mongodb.com/licensing/server-side-public-license/faq)), they are not technically open-source under the OSI definition. Why do we have this restriction if we love open source software so much? Since r2c offers a paid, hosted version at Semgrep.dev, it’s important for us to have this restriction so other companies like major cloud providers can’t just resell our rules as a competing service.
```

> If the LGPL + Common Clause license is an issue for you as a contributor, please give us feedback! r2c may be able to offer you semgrep-rules under a separate license on a case-by-case basis.

Reviewer: ```suggestion
We want to know if the LGPL + Common Clause license is an issue for you as a contributor! r2c may be able to offer you semgrep-rules under a separate license on a case-by-case basis.
```

> #### Is it ok to run Semgrep or the r2c rules on my work projects?

Reviewer: ```suggestion
#### Can I run Semgrep and its rules on my work projects?
```

> Yes! You should feel free to run your own private, offline Semgrep set up and use the r2c rules in it. Of course we’d love rules to be contributed back to the Semgrep Registry but you don’t have to. The Common Clause restriction only comes into effect if you are *selling* the rules in the semgrep-rules repo. If that’s the case, you’ll need to talk with r2c first to get permission.

Reviewer: ```suggestion
Yes! Semgrep is safe to run on your private code. The Common Clause restriction only comes into effect if you are *selling* the rules provided in the semgrep-rules repository. If that’s the case, you’ll need to talk with r2c first to get permission.
```

> #### I’m a security professional and want to use the semgrep-rules project with my clients as part of my paid services. Is that ok?

Reviewer: ```suggestion
#### Can I run Semgrep and its rules as part of my paid professional services for clients?
```

> Probably! If you have typical consulting service and running semgrep-rules is just a part of your assessments, that’s great—and of course feel free to refer your clients to the hosted [Semgrep.dev](https://semgrep.dev/). But if your entire service is about scanning code and you just want to charge people for running the semgrep-rules repo that r2c did the work to create and maintain, that’s something you’d need to reach out to us about.

Reviewer: ```suggestion
Probably! If you provide typical consulting services and running Semgrep and its rules is part of your assessments, that’s great — and of course we appreciate client references to the hosted [Semgrep.dev](https://semgrep.dev/). But if your entire service is about scanning code and you are charging clients for running the semgrep-rules repo that r2c did the work to create and maintain, that’s something you’d need to reach out to us about.
```
